### PR TITLE
Keypad

### PR DESCRIPTION
**Pull request by ctgreybeard**
This corrects some problems with keypads.

### CODE REVIEW

**Comment by ctgreybeard** [comment]
I have not done many pull requests and I'm not sure how to clean this one up. I would like to learn how to improve my workflow though.

This is working on my system with the six-button keypad.

**Comment by ctgreybeard** [comment]
I seem to have duplicated a bunch of the changes. Not sure how that happened.

**Comment by kuestess** [comment]
@ctgreybeard Checking to see if this is still in-progress - shows as draft so I can't merge.

**Comment by ctgreybeard** [comment]
Let me try again with this. I tried cleaning it up and only made it worse. I'll resubmit a better pull request and then close this one.
